### DISCUSSION TITLE
Github Discussion forum

### DISCUSSION THREAD

**Discussion by thomaspaulb** [OCA/odoo-community.org#77] [Polls]
Wow, so many nice features here. But does everyone think the same? Hence a poll.

**Comment by amh-mw** [comment]
I didn't get a notification when this poll went up. Is there any way to subscribe all? (Not just to this specific thread.)

**Reply by yajo** [reply]
Yes, go to https://github.com/OCA/odoo-community.org and subscribe to discussion notifications.

**Comment by Shide** [comment]
I think using mailing list and Github discussions is not a good idea because having 2 systems to do the same may be confusing.
I think Github discussions is more centralized and everyone can "See and reply" threads in the same screen.

**Reply by etobella** [reply]
But it has a problem, no notifications are received... Or at least I don't know how to do that :sob: 

Let's try with @yajo option....

**Reply by rousseldenis** [reply]
@etobella I can say I have this thread notifications, so, maybe a github profile setting ?

@Shide Mailing list is better to announce something, and discussion better to do the ... discussion on the particular subject to not pollute mailing list.

**Reply by yajo** [reply]
See https://odoo-community.org/groups/contributors-15/contributors-269341?mode=thread&date_begin=&date_end=

This is just an experiment for now. So we can have both options and see if it makes sense to keep both, or dump one of them.

> Mailing list is better to announce something

You can see that the forum has an announcements category: 

![imagen](https://user-images.githubusercontent.com/973709/224669729-dbe13935-afc4-4d9c-afac-d80b968b3b69.png)

**Comment by Shide** [comment]
Just in case something don't find subscribing to Discussions:
Go to Watch button in top right corner > Custom > Check discussions > Clic Apply

![image](https://user-images.githubusercontent.com/1162050/224649173-ab5bcd26-41a1-4f42-a40f-89a8fb35f3c1.png)
![image](https://user-images.githubusercontent.com/1162050/224649309-c8aae295-1296-4a55-945e-b3d453a5dbb2.png)

**Reply by yajo** [reply]
You can find those options in https://github.com/OCA/odoo-community.org

**Reply by sbidoul** [reply]
@yajo these are for the discussions in the odoo-community.org project. But how to subscribe to these OCA org-level discussions only?

**Reply by yajo** [reply]
It's really the same. The ORG-level discussions are just a mapping for one selected repo discussions in that ORG. See https://docs.github.com/en/discussions/quickstart#enabling-github-discussions-on-your-organization

**Comment by rvalyi** [comment]
hello, a concern I have with GitHub discussions is: many people will obviously use it to get free support.

At some point this is OK, but for the long time Odooers, you probably noticed our market already suffers some kind of "prostitution", specially with the recent years reshaping of the Odoo SA partner network around "money only" EE sale figures criteria. In more specific terms this is called a "market for lemons" https://saylordotorg.github.io/text_introduction-to-economic-analysis/s19-01-market-for-lemons.html#:~:text=The%20market%20for%20lemons%20refers,certification%20mitigate%20the%20lemons%20problem. and this is this typically how markets use to collapse when customers have less information about the product or service quality than the sellers (eg the Odoo experts).

Well obviously, having forums were people will expect free support will accelerate even more this "market for lemon" transition even in our OCA niche. 

Are we prepared for this? The answer may be yes and this might be the price to pay for scaling the OCA. But people expecting free support (and soon very low quality free support) on these forums will be the new normal like in any Magento or WordPress forums. And of course if you are the bad guy closing free support questions, you will be the one bad guy with a bad reputation...

So I'm not telling we should avoid GitHub Discussions, but I'm just making sure everybody is aware what will be the long term consequences.

**Reply by pedrobaeza** [reply]
Very good point. I would add that if not someone (me, hehe) being the bad guy closing the posts, then the alternative is to have hundreds of unanswered threads.

**Reply by rvalyi** [reply]
and the market for lemons typically happens when yet an other excellent contributor like you might be so pissed off to get its energy sucked away by free loaders that he abandons the ship, leaving it to an always decreasing quality. Something so classical when you follow open source projects leaders in the long term...

That's may be something to be discussed in the OCA board without a bad guy having to carry the responsabilty for this. My personal opinion is that this is inevitable in the long run, but may be the OCA ecosystem is still too fragile to be exposed to such new pressure and the moment has not come yet. BTW, this is very much why we decided to postpone GitHub Discussions for OCA/l10n-brazil untill the core module design got more robust because once a mass of random users will make huge pressures in these forums and support demagogic leaders, taking the hard design decisions will hardly be possible anymore...

**Reply by pedrobaeza** [reply]
Inside the reading you have linked, which is very interesting, there's a mention to a way to a measure for avoiding the flattening of the prices, like certifications. I think this is applicable to us, and we have already started at Odoo Spanish Association (AEOdoo) with that idea issuing a technical certification: https://www.aeodoo.org/certificacion-odoo

This is something to be considered at OCA level.

**Reply by thomaspaulb** [reply]
It is a good point, but isn't it easy to solve by requiring OCA membership in order to be able to start new topics?

(On forums, they have one "public" topic which is maintained by a bad guy, and the rest is readonly for all but members).

**Reply by pedrobaeza** [reply]
AFAIK, we can't model that in GitHub.

**Reply by legalsylvain** [reply]
> It is a good point, but isn't it easy to solve by requiring OCA membership in order to be able to start new topics?

:+1: If the idea is to move some discussion from mailing to github discussions. (and i think it's a good move, because all the mail regarding PSC with a lot of :+1: are not optimal), indeed, we can define the rule that only oca members can open new discussion.

If people are agree with that point, developping a little new feature in ocabot that closes automatically discussions from non members with a nice message like _"this forum is currently reserved for OCA members. Feel free to join the association to support it and have the possibility to post!"_ looks feasible.

**Reply by rvalyi** [reply]
+1 for such a measure at least temporarily.

Pedro, as for certifications and other market for lemon counter measures, well we can have them sure. But, open source projects also already carry a good deal of "quality signaling", like who makes the code, who fixes the bugs etc, all done in the open.

The real shoot in the feets of Odoo SA partnership system is to take all this and fake it's worth nothing vs being a good short term reseller telling any lie required to make a sale. It's also worth mentioning that many certified Odoo SA partners use fails projects massively despite their useless certification. So we can certainly have some serious OCA certification but it would also be good to avoid the same pitfalls...

**Reply by yajo** [reply]
IMHO think this proposal is an overkill.



> many people will obviously use it to get free support

This is an assumption, but not really a fact yet AFAIK. Even if it were a fact, is it a real problem?



> to have hundreds of unanswered threads

This is a forum, not an issue tracker. Open threads harm nobody. There's no need to close them. Just, don't answer energy suckers.



> requiring OCA membership in order to be able to start new topics?

Bad idea. Entering this community is hard enough already. And asking people to pay for making questions without any warranty on getting answers is like telling them: we only care about your money, not your contributions.

OTOH, if we make the forum more accessible, it'll be easier for other people to answer the questions we don't want to. That'd mean less weight in old contributors' shoulders.

A better idea, though, would be for OCA to start a program where contributors can subscribe either as support providers or as support customers. Then OCA can act as an intermediary providing the platform for such support. That would be better handled at [OCA's own Odoo](https://odoo-community.org) and something completely separate from being or not a member or being able or not to post a thread in a forum. It would be a way to finance the association and share that profit with the support providers.

**Reply by legalsylvain** [reply]
Hi thanks for your answer,

>  Open threads harm nobody. 

Well, for me it can harm.
 
- if the forum is reserved to OCA members, I will subscribe to it, and thus be kept informed, (via the github notification system), of new topics, and I will answer them.
- if the forum is open to everyone, it is very likely that many irrelevant topics will be opened. (See other odoo forum : https://www.odoo.com/forum/help-1) This will bring a lot of noise in the notifications, and I will unsubscribe from github notifications for this forum. As a result, I will be less kept up to date on topics that are relevant to OCA.

I had understood that the use of this forum was to replace the mailing lists, which are not very adapted to certain subjects, like : 
- "I would like to be PSC of repo X". "+1" / "+1" / "+1" / "+1" / "+1" 
- "split server repos into server-env / server-auth / ... . "+1" / "+1" / "-1" / "+1" 

The fact that non-OCA members participate in this topic does not seem vital to me.

**Comment by yajo** [comment]
El mar, 21 de mar de 2023 a las 02:40:50 AM, Sylvain LE GAL 
***@***.***> escribió:
> if the forum is open to everyone, it is very likely that many 
> irrelevant topics will be opened. (See other odoo forum : 
> <https://www.odoo.com/forum/help-1>) This will bring a lot of noise 
> in the notifications, and I will unsubscribe from github 
> notifications for this forum. As a result, I will be less kept up to 
> date on topics that are relevant to OCA.

There are better forum alternatives, such as discourse, which notify 
you with a digest of hottest topics. The digest can be weekly, 
monthly... depends on your taste. It also features gamification and 
other stuff.

However, that means an extra financial or maintenance burden, and 
that's why I suggested the github backend, which is plug and play. We 
can hope some day they improve this thing, or move to a better 
alternative.

Personally, I have that problem you mention already with issues and 
PRs. I have a specific filter in my inbox for threads that mention me, 
where I pay more attention. It works for discussions too FWIW.

**Comment by l-arnold** [comment]
A bit late to the discussion, but it seems pertinent.
If this is the place for "discussions" I don't see the harm at all.  If there is another place, perhaps it should be anounced.
Deciphering info, usage, and methods on OCA is often difficult and the "Community" word in OCA would benefit from just such a central discussion/forum space.
+1 from here.
